From 9a17d28d0b86b38a0b1a4e361cadbd0cb3628953 Mon Sep 17 00:00:00 2001
From: Terry Lv <r65388@freescale.com>
Date: Fri, 27 Nov 2009 15:47:24 +0800
Subject: [PATCH] ENGR00118801: Unknown information when booting from SPI-NOR.

No 'boot up from' information when booting from NOR-SPI.

Signed-off-by: Terry Lv <r65388@freescale.com>
---
 board/freescale/mx51_3stack/mx51_3stack.c |    2 +-
 board/freescale/mx51_bbg/mx51_bbg.c       |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/freescale/mx51_3stack/mx51_3stack.c b/board/freescale/mx51_3stack/mx51_3stack.c
index 72df29c..2dc6b2d 100644
--- a/board/freescale/mx51_3stack/mx51_3stack.c
+++ b/board/freescale/mx51_3stack/mx51_3stack.c
@@ -72,7 +72,7 @@ static inline void setup_boot_device(void)
 		{
 			uint soc_sbmr = readl(SRC_BASE_ADDR + 0x4);
 			uint bt_mem_ctl = soc_sbmr & 0x00000003;
-			uint bt_mem_type = (soc_sbmr & 0x000000C0) >> 7;
+			uint bt_mem_type = (soc_sbmr & 0x00000180) >> 7;
 
 			switch (bt_mem_ctl) {
 			case 0x3:
diff --git a/board/freescale/mx51_bbg/mx51_bbg.c b/board/freescale/mx51_bbg/mx51_bbg.c
index 0a4bc51..e5287d0 100644
--- a/board/freescale/mx51_bbg/mx51_bbg.c
+++ b/board/freescale/mx51_bbg/mx51_bbg.c
@@ -75,7 +75,7 @@ static inline void setup_boot_device(void)
 		{
 			uint soc_sbmr = readl(SRC_BASE_ADDR + 0x4);
 			uint bt_mem_ctl = soc_sbmr & 0x00000003;
-			uint bt_mem_type = (soc_sbmr & 0x000000C0) >> 7;
+			uint bt_mem_type = (soc_sbmr & 0x00000180) >> 7;
 
 			switch (bt_mem_ctl) {
 			case 0x3:
-- 
1.5.4.4

